### DISCUSSION TITLE
EKS 노드그룹 생성 실패 관련 문의

### DISCUSSION THREAD

**Discussion by Virusuki** [tedilabs/fastcampus-devops#75] [Q&A] [answered]
안녕하세요, 박병진 강사님 좋은 강의 감사드립니다.

[추가파트] Kubernetes on AWS EKS 에 강의 따라 실습하는데,,, 클러스터 구성에서 노드그룹 생성 시 실패 이슈가 발생하고 있습니다.

그리고, 말씀하신 사전 준비의 
3가지 정책 설정했고, aws cli를 통해 kubectl cluster-info 를 통해 제어플레인도 확인되었습니다.
- AmazonEKSWorkerNodePolicy
- AmazonEKS_CNI_Policy
- AmazonEC2ContainerRegistryReadOnly 

해당 이슈에 대해 문의드립니다. ㅠㅠ

![eks노드생성실패1](https://user-images.githubusercontent.com/70243256/178401510-7ff0824a-504e-4409-9fa0-26c6a11bfc57.PNG)

**Comment by Virusuki** [comment, marked as answer]
아참, 강사님 VPC 셋팅은 테라폼을 이용하여 셋팅했는데,, 수동으로 어떻게 VPC를 구성하면 될지 알수있을까요??..
저는 VPC 1개 (CIDR-10.0.0.0/16) 으로 했고,
서브넷 2개 만들어서 IPv4 CIDR 10.0.1.0/24 와 10.0.0.0/24 이렇게 서브넷 2개를 생성하여 네트워킹을 설정했었습니다.

**Comment by posquit0** [comment]
안녕하세요~ 답변이 늦어 죄송합니다.
우선, 노드그룹이 클러스터 조인에 실패하는 대부분의 이유는 **Node <-> Control Plane** 간의 네트워크 통신에 문제가 있기 때문입니다.
VPC를 강의에서 제공한 테라폼 코드가 아니라 직접 구성한 경우 아래 내용을 살펴보시길 권장드립니다.

- NAT Gateway 구성: EKS 노드그룹이 생성되는 VPC 서브넷의 라우트 테이블 상에 NAT GW로의 라우팅 룰이 존재하나요?
- NACL 규칙 검토: 노드<->노드, 노드 <->컨트롤 플레인 상 필요로 하는 네트워크 구간이 허용되어 있나요?
- 보안그룹(Security Group) 검토
  - Control Plane Security Group: 노드로부터의 API 접근 통신이 허용되어 있나요?
  - Node Security Group: 컨트롤플레인으로부터 접근이 허용되어 있나요? 컨트롤플레인으로의 통신이 허용되어 있나요?


### 참고자료

아래 공식 문서 상에서 어떠한 네트워크 조건들이 갖추어져야하는지 보다 자세히 나와있습니다. :)

- [EKS Network Requirements and Considerations](https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html)

**Comment by Virusuki** [comment]
해결되었습니다. 감사합니다. :)
